lib/chef/provider/package/easy_install.rb

```diff
@@ -94,10 +93,10 @@ def load_current_resource
 
           if package_version == @new_resource.version
             Chef::Log.debug("#{@new_resource} at version #{@new_resource.version}")
-            @current_resource.version(@new_resource.version)
+            @current_resource.version(@new_resource.version) if @new_resource.version
```

Comment:
The argument that I'd have here is that nil is perfectly valid, and is the default that we must have on the new_resource in order to indicate that the user had no preference.  We are copying those values to the current_resource, and I think you're getting fairly lost in philosophical semantics here.  You should add 'nil' to the allowed values, and then not only don't you need to guard every single one of these, but the user can 'un-set' a value back to the cleared default.  Otherwise there's no way to clear something once you've set it (which is hugely annoying in spec tests for one thing).